tests/test_entities.rb

```diff
@@ -0,0 +1,131 @@
+    person = Person.new(20, 'John')
+    expect(person.name).to eq('John')
+  end
+  it 'should have an age' do
+    person = Person.new(20, 'John')
+    expect(person.age).to eq(20)
+  end
+  it 'should have a parent permission' do
+    person = Person.new(20, 'John')
+    expect(person.parent_permission).to eq(true)
+  end
+end
+
+RSpec.describe Student do
+  it 'should be a person' do
+    student = Student.new(20, 'John', parent_permission: true)
+    expect(student.class).to eq(Student)
+  end
+  it 'should have a name' do
+    student = Student.new(20, 'Fernando', parent_permission: true)
+    expect(student.name).to eq('Fernando')
+  end
+  it 'should have an age' do
+    student = Student.new(20, 'John', parent_permission: true)
+    expect(student.age).to eq(20)
+  end
+end
+
+RSpec.describe Teacher do
+  it 'should be a person' do
+    teacher = Teacher.new(20, 'Ruby', 'John')
+    expect(teacher.class).to eq(Teacher)
+  end
+  it 'should have a name' do
+    teacher = Teacher.new(20, 'Ruby', 'John')
+    expect(teacher.name).to eq('John')
+  end
+  it 'should have an age' do
+    teacher = Teacher.new(20, 'Ruby', 'John')
+    expect(teacher.age).to eq(20)
+  end
+  it 'should have a specialization' do
+    teacher = Teacher.new(20, 'Ruby', 'John')
+    expect(teacher.specialization).to eq('Ruby')
+  end
+end
+
+RSpec.describe Rental do
+  it 'should be a rental' do
+    rental = Rental.new('2020-01-01', Book.new('Ruby', 'John'), Person.new(20, 'John'))
+    expect(rental.class).to eq(Rental)
+  end
+  it 'should have a date' do
+    rental = Rental.new('2020-01-01', Book.new('Ruby', 'John'), Person.new(20, 'John'))
+    expect(rental.date).to eq('2020-01-01')
+  end
+end
+
+RSpec.describe Decorator do
+  it 'should have nameable' do
+    nameable = Decorator.new(Person.new(20, 'John'))
+    expect(nameable.class).to eq(Decorator)
+  end
+
+  it 'should have correct name' do
+    correct_name = Decorator.new(Person.new('John', 'John'))
+    expect(correct_name.class).to eq(Decorator)
+  end
+end
+
+RSpec.describe CapitalizedDecorator do
+  it 'should start with a capital letter' do
+    book = CapitalizedDecorator.new(Book.new('Gifted', 'Ben'))
+    expect(book.class).to eq(CapitalizedDecorator)
+  end
+end
+
+RSpec.describe TrimmerDecorator do
+  it 'should trim the length' do
+    trimmer = TrimmerDecorator.new(Student.new(20, 'Johnmarystewart', parent_permission: true))
+    expect(trimmer.correct_name).to eq('Johnmaryst')
+  end
+end
+
+RSpec.describe Classroom do
+  it 'should have a label' do
+    classroom = Classroom.new('2S')
+    expect(classroom.label).to eq('2S')
+  end
+
+  it 'should have students' do
+    classroom = Classroom.new('2S')
+    expect(classroom.students).to eq([])
+  end
+
+  it 'should accept students' do
+    classroom = Classroom.new('2S')
+    student = Student.new(20, 'John', parent_permission: true)
+    classroom.add_student(student)
+    expect(classroom.students).to eq([student])
+  end
+end
```

Comment:
- [ ] Nice job :clap: on the Writing unit test all of them are passing, however, I would suggest you separate tests into files, so that each file contains specific logic for a given `class`(eg: `spec/person_spec.rb` this file would contain only tests of the `Person` class)

Reply:
@andersonlebon Thank you for the observation. I'll do that change here and keep it in mind for my next projects 👍